core/src/main/java/org/frankframework/pipes/ForPipe.java

```diff
@@ -0,0 +1,122 @@
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.doc.ElementType;
+import org.frankframework.doc.ElementType.ElementTypes;
+import org.frankframework.stream.Message;
+
+/**
+ * ForPipe is a wrapper to use another pipe a fixed number of times. This can be accomplished by something like:
+ *
+ * <pre>{@code
+ * 		<ForPipe name="forPipe" incrementSessionKey="i" startAt="0" max="10">
+ * 		 	<Forward name="stop" path="EXIT" />
+ * 		 	<Forward name="continue" path="echoPipe"/>
+ * 		</ForPipe>
+ *
+ * 		<EchoPipe name="echoPipe" getInputFromSessionKey="i">
+ * 		 	<Forward name="success" path="forPipe"/>
+ * 		</EchoPipe>
+ * }</pre>
+ * <p>
+ * This should call the echoPipe for i=0 until i=10.
+ *
+ * @author evandongen
+ */
+@ElementType(ElementTypes.ITERATOR)
+public class ForPipe extends FixedForwardPipe {
+
+	static final String STOP_FORWARD_NAME = "stop";
+	static final String CONTINUE_FORWARD_NAME = "continue";
+	private @Getter String incrementSessionKey = "i";
+	private @Getter int startAt = 0;
+	private @Getter Integer max = null;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		// Mandatory parameters
+		if (max == null) {
+			throw new ConfigurationException("Value for 'max' is mandatory to break out of the for loop pipe");
+		}
+
+		// Mandatory forwards
+		PipeForward stopForward = findForward(STOP_FORWARD_NAME);
+		PipeForward continueForward = findForward(CONTINUE_FORWARD_NAME);
+
+		if (stopForward == null) {
+			throw new ConfigurationException("has no forward with name [" + STOP_FORWARD_NAME + "]");
+		}
+
+		if (continueForward == null) {
+			throw new ConfigurationException("has no forward with name [" + CONTINUE_FORWARD_NAME + "]");
+		}
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
+		// Get the value from the session, default to the startAt value
+		int i = session.get(incrementSessionKey, startAt);
+
+		// If i < max, increment and continue
+		if (i < max) {
+			PipeForward continueForward = findForward(CONTINUE_FORWARD_NAME);
+
+			session.put(incrementSessionKey, ++i);
+
+			return new PipeRunResult(continueForward, message);
+		}
+
+		// Else, remove the session key and forward to the stop forward
+		session.remove(incrementSessionKey);
+		return new PipeRunResult(findForward(STOP_FORWARD_NAME), message);
+	}
+
+	/**
+	 * Sets the session key which holds the value to be incremented.
+	 *
+	 * @ff.default i
+	 */
+	public void setIncrementSessionKey(String incrementSessionKey) {
+		this.incrementSessionKey = incrementSessionKey;
+	}
+
+	/**
+	 * Starts counting at this value.
+	 *
+	 * @ff.default 0
+	 */
+	public void setStartAt(int startAt) {
+		this.startAt = startAt;
+	}
+
+	/**
+	 * Break from the loop when incrementSessionKey equals this value
+	 */
+	public void setMax(int max) {
```

Comment:
The setter for `setMax` has a different type than the actual `max` field. Perhaps that is why the test breaks on this.

Reply:
Agreed, this should not fail the test because of autoboxing. I've modified the setter to receive an Integer for now.